# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLParsers.cpp
# Commit: ef6d81f0b298
# Full Hash: ef6d81f0b2989624a9918ec5d790dc2745db20ef
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2018-11-15 09:52:43
# Regressor Bug: 1495313
# File Overlap Count: 1
# Description:
#   Bug 1495313 - Allow password with empty username in URLs r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D11254
# ==============================================================================

diff -r 06b12fd41ff1 -r ef6d81f0b298 netwerk/base/nsURLParsers.cpp
--- a/netwerk/base/nsURLParsers.cpp	Wed Nov 14 19:37:42 2018 +0000
+++ b/netwerk/base/nsURLParsers.cpp	Wed Nov 14 19:10:42 2018 +0000
@@ -535,10 +535,6 @@
     const char *p = (const char *) memchr(userinfo, ':', userinfoLen);
     if (p) {
         // userinfo = <username:password>
-        if (p == userinfo) {
-            // must have a username!
-            return NS_ERROR_MALFORMED_URI;
-        }
         SET_RESULT(username, 0, p - userinfo);
         SET_RESULT(password, p - userinfo + 1, userinfoLen - (p - userinfo + 1));
     }